app/controlplane/pkg/policies/policyprovider.go

```diff
@@ -33,51 +34,84 @@ type PolicyProvider struct {
 
 type ProviderResponse struct {
 	Policy map[string]any `json:"policy"`
+	Digest string         `json:"digest"`
+}
+
+type PolicyReference struct {
+	URL    string
+	Digest string
 }
 
 // Resolve calls the remote provider for retrieving a policy
-func (p *PolicyProvider) Resolve(policyName string, token string) (*schemaapi.Policy, error) {
+func (p *PolicyProvider) Resolve(policyName string, token string) (*schemaapi.Policy, *PolicyReference, error) {
 	if policyName == "" || token == "" {
-		return nil, fmt.Errorf("both policyname and token are mandatory")
+		return nil, nil, fmt.Errorf("both policyname and token are mandatory")
 	}
 
 	req, err := http.NewRequest("GET", fmt.Sprintf("%s/%s", p.host, policyName), nil)
 	if err != nil {
-		return nil, fmt.Errorf("error creating policy request: %w", err)
+		return nil, nil, fmt.Errorf("error creating policy request: %w", err)
 	}
 
 	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
 
 	// make the request
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("error executing policy request: %w", err)
+		return nil, nil, fmt.Errorf("error executing policy request: %w", err)
 	}
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("expected status code 200 but got %d", resp.StatusCode)
+		return nil, nil, fmt.Errorf("expected status code 200 but got %d", resp.StatusCode)
 	}
 
 	resBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return nil, fmt.Errorf("error reading policy response: %w", err)
+		return nil, nil, fmt.Errorf("error reading policy response: %w", err)
 	}
 
 	// unmarshall response
 	var response ProviderResponse
 	if err := json.Unmarshal(resBytes, &response); err != nil {
-		return nil, fmt.Errorf("error unmarshalling policy response: %w", err)
+		return nil, nil, fmt.Errorf("error unmarshalling policy response: %w", err)
+	}
+
+	ref, err := p.resolveRef(policyName, response.Digest)
+	if err != nil {
+		return nil, nil, fmt.Errorf("error resolving policy reference: %w", err)
 	}
 
 	// extract the policy payload from the query response
 	jsonPolicy, err := json.Marshal(response.Policy)
 	if err != nil {
-		return nil, fmt.Errorf("error marshalling policy response: %w", err)
+		return nil, nil, fmt.Errorf("error marshalling policy response: %w", err)
 	}
 
 	// unmarshall the payload to the known protobuf message for policies
 	var res schemaapi.Policy
 	if err := protojson.Unmarshal(jsonPolicy, &res); err != nil {
-		return nil, fmt.Errorf("error unmarshalling policy response: %w", err)
+		return nil, nil, fmt.Errorf("error unmarshalling policy response: %w", err)
 	}
-	return &res, nil
+
+	return &res, ref, nil
+}
+
+func (p *PolicyProvider) resolveRef(policyName, digest string) (*PolicyReference, error) {
+	// Extract hostname from the policy provider URL
+	uri, err := url.Parse(p.host)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing policy provider URL: %w", err)
+	}
+
+	if uri.Host == "" {
```

Comment:
I wonder if we should store the full URL, not only the host.

Reply:
the reason of not storing the full URL is because to me, the implementation of how this policy is retrieved is not that relevant, in fact it might change in the future.

I was thinking of the OCI registry example, when you use an identifier such as i.e gcr.dev/repo/image, this will get translated to API calls to `v2/manifests` and `v2/blobs`. So this is the same. The client, the provider will know how to retrieve the policy. 

WDYT?

Reply:
Yes, it makes sense 👍, thanks